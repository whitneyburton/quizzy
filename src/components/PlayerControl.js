import React, { Component } from 'react';
import '../styles/main.scss';

export default class PlayerControl extends Component {
  constructor() {
    super();
    this.state = {

    }
  }

  passCategory = () => {
    if (JSON.parse(localStorage.getItem('incorrectFlashcardsStorage'))) {
      let clickedCategory = 'Incorrect Only';
      this.props.updateCategory(clickedCategory);
    } else {
      let clickedCategory = 'No study cards yet - Keep guessing!'
      this.props.updateCategory(clickedCategory);      
    }
  }

  render() {
    let { incorrectFlashcards, deleteAllStorage } = this.props;
    return (
      <div className="PlayerControl">
        <h2 className="player-options">Player Options</h2>
        <button
          onClick={this.passCategory}          
          className="view-incorrect-button buttons"
          type="button">View Study List ({incorrectFlashcards.length})</button>
        <button
          onClick={deleteAllStorage}                    
          className="reset-button buttons"
          type="button">Reset Quizzy</button>
      </div>
    );
  }
}