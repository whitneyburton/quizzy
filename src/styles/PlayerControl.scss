.PlayerControl {
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 81vh;
  background-image: linear-gradient($whitesmoke, #FFF);
  width: 30%;
  margin: auto;

  .player-options {
    padding-top: 14%;
    color: #2d2d2d;
  }

  .view-incorrect-button,
  .reset-button {
    width: 45%;
    padding: 20px;
    margin-top: 35px;
  }
}


@media screen and (max-width: 755px) {
  .PlayerControl {
    width: 100%;
    height: auto;
    padding-top: 2%;

    .player-options {
      padding-top: 3%;
    }
  
    .view-incorrect-button,
    .reset-button {
      width: 45%;
      padding: 10px;
      margin: 10px 0px 10px 0px;
    }
  }
}