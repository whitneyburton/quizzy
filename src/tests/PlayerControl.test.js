import React from 'react';
import PlayerControl from '../Components/PlayerControl';
import { shallow } from 'enzyme';

describe('PlayerControl', () => {
  let wrapper;
  const filteredCards = [{}, {}, {}]
  const updateCategoryMock = jest.fn();
  const removeStorageMock = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <PlayerControl
        filteredCards={filteredCards}
        updateCategory={updateCategoryMock}
        removeStorage={removeStorageMock}
      />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})