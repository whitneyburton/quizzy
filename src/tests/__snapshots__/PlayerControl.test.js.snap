// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerControl should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlayerControl
    filteredCards={
      Array [
        Object {},
        Object {},
        Object {},
      ]
    }
    removeStorage={[MockFunction]}
    updateCategory={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Your Stats:
        </h2>,
        <p
          className="correct-answers stats"
        >
          CORRECT:
          <span>
             
            0
          </span>
        </p>,
        <p
          className="incorrect-answers stats"
        >
          INCORRECT:
          <span>
             
            0
          </span>
        </p>,
        <p
          className="unanswered-answers stats"
        >
          UNANSWERED:
          <span>
             
            0
          </span>
        </p>,
        <p
          className="accuracy stats"
        >
          ACCURACY:
          <span>
             
            0
          </span>
          %
        </p>,
        <button
          className="view-incorrect-button buttons"
          onClick={[Function]}
          type="button"
        >
          View Study List
        </button>,
        <button
          className="reset-button buttons"
          onClick={[MockFunction]}
          type="button"
        >
          Reset Quizzy
        </button>,
      ],
      "className": "PlayerControl",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Your Stats:",
        },
        "ref": null,
        "rendered": "Your Stats:",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "CORRECT:",
            <span>
               
              0
            </span>,
          ],
          "className": "correct-answers stats",
        },
        "ref": null,
        "rendered": Array [
          "CORRECT:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                0,
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              0,
            ],
            "type": "span",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "INCORRECT:",
            <span>
               
              0
            </span>,
          ],
          "className": "incorrect-answers stats",
        },
        "ref": null,
        "rendered": Array [
          "INCORRECT:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                0,
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              0,
            ],
            "type": "span",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "UNANSWERED:",
            <span>
               
              0
            </span>,
          ],
          "className": "unanswered-answers stats",
        },
        "ref": null,
        "rendered": Array [
          "UNANSWERED:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                0,
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              0,
            ],
            "type": "span",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "ACCURACY:",
            <span>
               
              0
            </span>,
            "%",
          ],
          "className": "accuracy stats",
        },
        "ref": null,
        "rendered": Array [
          "ACCURACY:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                0,
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              0,
            ],
            "type": "span",
          },
          "%",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "View Study List",
          "className": "view-incorrect-button buttons",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "View Study List",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Reset Quizzy",
          "className": "reset-button buttons",
          "onClick": [MockFunction],
          "type": "button",
        },
        "ref": null,
        "rendered": "Reset Quizzy",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Your Stats:
          </h2>,
          <p
            className="correct-answers stats"
          >
            CORRECT:
            <span>
               
              0
            </span>
          </p>,
          <p
            className="incorrect-answers stats"
          >
            INCORRECT:
            <span>
               
              0
            </span>
          </p>,
          <p
            className="unanswered-answers stats"
          >
            UNANSWERED:
            <span>
               
              0
            </span>
          </p>,
          <p
            className="accuracy stats"
          >
            ACCURACY:
            <span>
               
              0
            </span>
            %
          </p>,
          <button
            className="view-incorrect-button buttons"
            onClick={[Function]}
            type="button"
          >
            View Study List
          </button>,
          <button
            className="reset-button buttons"
            onClick={[MockFunction]}
            type="button"
          >
            Reset Quizzy
          </button>,
        ],
        "className": "PlayerControl",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Your Stats:",
          },
          "ref": null,
          "rendered": "Your Stats:",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "CORRECT:",
              <span>
                 
                0
              </span>,
            ],
            "className": "correct-answers stats",
          },
          "ref": null,
          "rendered": Array [
            "CORRECT:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  0,
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                0,
              ],
              "type": "span",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "INCORRECT:",
              <span>
                 
                0
              </span>,
            ],
            "className": "incorrect-answers stats",
          },
          "ref": null,
          "rendered": Array [
            "INCORRECT:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  0,
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                0,
              ],
              "type": "span",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "UNANSWERED:",
              <span>
                 
                0
              </span>,
            ],
            "className": "unanswered-answers stats",
          },
          "ref": null,
          "rendered": Array [
            "UNANSWERED:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  0,
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                0,
              ],
              "type": "span",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "ACCURACY:",
              <span>
                 
                0
              </span>,
              "%",
            ],
            "className": "accuracy stats",
          },
          "ref": null,
          "rendered": Array [
            "ACCURACY:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  0,
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                0,
              ],
              "type": "span",
            },
            "%",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "View Study List",
            "className": "view-incorrect-button buttons",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "View Study List",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Reset Quizzy",
            "className": "reset-button buttons",
            "onClick": [MockFunction],
            "type": "button",
          },
          "ref": null,
          "rendered": "Reset Quizzy",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
